src/python/bcc/table.py

```diff
@@ -523,8 +524,24 @@ def open_perf_buffer(self, callback, page_cnt=8, lost_cb=None):
             self._open_perf_buffer(i, callback, page_cnt, lost_cb)
 
     def _open_perf_buffer(self, cpu, callback, page_cnt, lost_cb):
-        fn = _RAW_CB_TYPE(lambda _, data, size: callback(cpu, data, size))
-        lost_fn = _LOST_CB_TYPE(lambda lost: lost_cb(lost)) if lost_cb else ct.cast(None, _LOST_CB_TYPE)
+        def raw_cb_(_, data, size):
+            try:
+                callback(cpu, data, size)
+            except IOError, e:
+                if e.errno == errno.EPIPE:
+                    exit()
+                else:
+                    raise e
```

Comment:
The callback function can be arbitrary user function here. Why you single out EPIPE errno from
another other exceptions?

Reply:
If the developer makes a mistake in his callback function (say, write to a file on which she doesn't have permissions), we want to display the usual traceback, no?

Reply:
Any exception should already have a traceback here?

Reply:
Yes. If I exit on every exception (or ignore some), these tracebacks won't be displayed.

Reply:
So why the exit for errno.EPIPE? If you have an example of callback function and/or lost_cb function, probably it is easier to try and understand your point?

Reply:
The goal is to properly exit when receiving EPIPE or SIGPIPE as described in #1010.
If you take the example from #1010, when running `./opensnoop.py | head`, it will display 10 lines then exception tracebacks for each new line without exiting because of the EPIPE Exception. With the present pull request it will properly exit (call cleanup and not display a traceback) on the first EPIPE Exception.